File: gateway/src/main/java/com/beat/gateway/jwt/internal/JwtProperties.java
package com.beat.gateway.jwt.internal;

import java.time.Instant;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "jwt")
public record JwtProperties(
	String secret,
	String legacySecret,
	long accessTokenExpireTime,
	long refreshTokenExpireTime,
	String keyId,
	Instant legacyAccessTokenVerifyUntil
) {
}

File: support/security/src/main/kotlin/com/beat/support/security/jwt/internal/JwtProperties.kt
package com.beat.support.security.jwt.internal

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Positive
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@Validated
@ConfigurationProperties(prefix = "jwt")
data class JwtProperties(
    @field:NotBlank val secret: String,
    @field:Positive val accessTokenExpireTime: Long,
    @field:Positive val refreshTokenExpireTime: Long,
    @field:NotBlank val keyId: String,
)
